packages/blocks/src/_common/adapters/plain-text.ts

```diff
@@ -232,6 +232,7 @@ export class PlainTextAdapter extends BaseAdapter<PlainText> {
       return null;
     }
     payload.file = payload.file.replaceAll('\r', '');
+    payload.file = payload.file.replace(/\n$/, '');
```

Comment:
I think the reason is in the exporter side, not here. @fourdim PTAL.

Reply:
yeah, i think so too ,  here is the export side in `/packages/blocks/src/_common/adapters/plain-text.ts`
```ts
  private async _traverseSnapshot(
    snapshot: BlockSnapshot
  ): Promise<{ mixtext: string }> {
    ...
    walker.setEnter(o => {
      const text = (o.node.props.text ?? { delta: [] }) as {
        delta: DeltaInsert[];
      };
      switch (o.node.flavour) {
        ...
        case 'affine:paragraph': {
          buffer += text.delta.map(delta => delta.insert).join('');
          buffer += '\n';
          break;
        }
 ...
```
but i think change here will affect a lot , so i only change the adapter

Reply:
No, changing the adapter will affect a lot.

> It's not about adding an extra newline at the end of a file, it's about not removing the newline that should be there.

> The C standard specifies that a C file should end with a newline (C11, 5.1.1.2, 2.) and that a last line without a newline yields undefined behavior (C11, J.2, 2nd item).

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

Reply:
No, changing this adapter will affect a lot.

> It's not about adding an extra newline at the end of a file, it's about not removing the newline that should be there.

> The C standard specifies that a C file should end with a newline (C11, 5.1.1.2, 2.) and that a last line without a newline yields undefined behavior (C11, J.2, 2nd item).

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file